dubbo-user-book-en/references/registry/zookeeper.md

```diff
@@ -1,13 +1,13 @@
 # Zookeeper Registry Server
 
-[Zookeeper](http://zookeeper.apache.org) is the child project of apache hadoop. Since it offers tree-like directory service and supports change notification, it's suitable to use it as dubbo's registry server. It's a field-proven product, therefore it's recommended to use it in the production environment. [^1]
+[Zookeeper](http://zookeeper.apache.org) is a child project of [Apache Hadoop](http://hadoop.apache.org). Since it offers tree-like directory service and supports change notification, it's suitable to use it as dubbo's registry server. It's a field-proven product, therefore it's recommended to use it in the production environment. [^1]
 
 ![/user-guide/images/zookeeper.jpg](../../sources/images/zookeeper.jpg)
 
 Description on registration procedure:
 
 * When service provider boots up: write service URL address under directory `/dubbo/com.foo.BarService/providers`
-* When service consumer boots up: subscribe to `/dubbo/com.foo.BarService/providers` for provider's URL addresses. At the same time, write consumer's URL address under `/dubbo/com.foo.BarService/providers`.
+* When service consumer boots up: subscribe to `/dubbo/com.foo.BarService/providers` for provider's URL addresses. At the same time, write consumer's URL address under `/dubbo/com.foo.BarService/consumers`.
```

Comment:
I guess you meant 'consumers' instead of 'providers'

Reply:
The prefix for writing the key is obtained by reading the group attribute. eg: `/dubbo/com.foo.BarService/providers` ,  for example, the service group is test, the key is changed to '/test/com.foo.BarService/providers' or '/test/com.foo.BarService/consumers'.

Reply:
Hi @zonghaishang ,

thank you for your reply but my question was about something different. In short I think that the current documentation contains a typo. The actual documentation says:

> When service consumer boots up: subscribe to `/dubbo/com.foo.BarService/providers` for provider's URL addresses. At the same time, write consumer's URL address under '/dubbo/com.foo.BarService/**providers**'.

but I think it should be:

> When service consumer boots up: subscribe to `/dubbo/com.foo.BarService/providers` for provider's URL addresses. At the same time, write consumer's URL address under '/dubbo/com.foo.BarService/**consumers**'.

(i.e. 'consumers' instead of 'providers' at the end of the paragraph)

Reply:
@zonghaishang Your complementary comments can be added by another pull request :)